sklearn/manifold/_t_sne.py

```diff
@@ -823,6 +848,9 @@ def _fit(self, X, skip_num_points=0):
             pca = PCA(n_components=self.n_components, svd_solver='randomized',
                       random_state=random_state)
             X_embedded = pca.fit_transform(X).astype(np.float32, copy=False)
+            # PCA is rescaled so that PC1 has standard deviation 1e-4 which is
+            # the default value for random initialization. See issue #18018.
+            X_embedded = X_embedded / np.std(X_embedded[:, 0]) * 1e-4
```

Comment:
I am not sure if we can consider this change a bugfix, so we might also need a proper future warning cycle here.

Reply:
You are right -- this is not a bugfix. All right, I made this a future warning. Take a look if it's thrown in the correct place.